server/routes/shared_api/autocomplete/helpers.py

```diff
@@ -71,97 +70,72 @@ def execute_maps_request(query: str, language: str) -> Dict:
-  """Computes a 'score' based on the matching words in two strings.
+def get_match_score(e: Dict) -> float:
+  """Computes a 'score' based on the matching words in two strings. Lowest
+  score is best match.
   Returns:
     Float score."""
+  match_string = e['matched_query']
+  name = e['description']
   rgx = re.compile(r'\s+')
   words_in_name = re.split(rgx, name)
   words_in_str1 = re.split(rgx, match_string)
 
   score = 0
+  start_index = 0
   for str1_word in words_in_str1:
     str1_word = str1_word.lower()
-    for name_word in words_in_name:
+    for idx, name_word in enumerate(words_in_name):
+      if idx < start_index:
+        continue
+
       name_word = name_word.lower()
       if str1_word == name_word:
-        score += 1
+        start_index = idx + 1
+        score -= 1
         break
       elif str1_word in name_word:
-        score += 0.5
+        start_index = idx + 1
+        score -= 0.5
         break
       else:
-        score -= 1
+        score += 1
 
   return score
 
 
-def find_best_match(name: str, string1: str, string2: str) -> str:
-  """Finds the best match between string1 and string2 for name. We use a very
-  simple algorithm based on approximate accuracy.
-  Returns:
-    String that is the better match.
-  """
-
-  # Note that this function is implemented to find the best "matched_query", when the same response
-  # is found multiple times.
-  # For example:
-  #   name: "California, USA"
-  #   string1: "Of Calif"
-  #   string2: "Calif"
-  # should return "Calif" as a better match.
-  score1 = get_match_score(name, string1)
-  score2 = get_match_score(name, string2)
-
-  if score2 > score1:
-    return string2
-
-  return string1
-
-
 def predict(queries: List[str], lang: str) -> List[Dict]:
   """Trigger maps prediction api requests and parse the output. Remove duplication responses and limit the number of results.
   Returns:
       List of json objects containing predictions from all queries issued after deduping.
   """
-  responses = []
-  place_ids = set()
-  duplicates = {}
-
+  all_responses = []
   for query in queries:
     predictions_for_query = execute_maps_request(query, lang)['predictions']
 
     for pred in predictions_for_query:
-      pred['matched_query'] = query
-      if pred['place_id'] not in place_ids:
-        place_ids.add(pred['place_id'])
-        responses.append(pred)
-      else:
-        if pred['place_id'] in duplicates:
-          # find best match
-          # print("Second dupe.")
-          bm = find_best_match(pred['description'],
-                               duplicates[pred['place_id']], query)
-          # print("BM won: ")
-          # print(bm)
-          duplicates[pred['place_id']] = bm
-        else:
-          # print("We're just getting our first dupe.")
-          duplicates[pred['place_id']] = query
-
-      if len(responses) >= RESPONSE_COUNT_LIMIT:
-        # prevent new loop to iterate through next answer.
-        break
+      result_pred = {}
```

Comment:
Consider making this a dataclass in a follow-up

Reply:
Yes still on my todo list!